packages/runner-shared/src/event-manager.js

```diff
@@ -500,6 +518,87 @@ export const eventManager = {
+      // Sync stable if the expected origin has loaded.
+      // Only listen to window load events from the most recent secondary origin, This prevents nondeterminism in the case where we redirect to an already
+      // established spec bridge, but one that is not the current or next cy.origin command.
+      if (cy.state('latestActiveOriginPolicy') === originPolicy) {
+        // We remain in an anticipating state until either a load even happens or a timeout.
+        cy.state('autOrigin', cy.state('autOrigin', cors.getOriginPolicy(url)))
+        cy.isAnticipatingCrossOriginResponseFor(undefined)
+        cy.isStable(true, 'load')
+        // Prints out the newly loaded URL
+        Cypress.emit('internal:window:load', { type: 'cross:origin', url })
+        // Re-broadcast to any other specBridges.
+        Cypress.primaryOriginCommunicator.toAllSpecBridges('window:load', { url })
+      }
+    })
+
+    Cypress.primaryOriginCommunicator.on('before:unload', () => {
+      // We specifically don't call 'cy.isStable' here because we don't want to inject another load event.
+      // Unstable is unstable regardless of where it initiated from.
+      cy.state('isStable', false)
+      // Re-broadcast to any other specBridges.
+      Cypress.primaryOriginCommunicator.toAllSpecBridges('before:unload')
+    })
+
+    Cypress.primaryOriginCommunicator.on('expect:origin', (originPolicy) => {
+      localBus.emit('expect:origin', originPolicy)
+    })
+
+    Cypress.primaryOriginCommunicator.on('viewport:changed', (viewport, originPolicy) => {
+      const callback = () => {
+        Cypress.primaryOriginCommunicator.toSpecBridge(originPolicy, 'viewport:changed:end')
+      }
+
+      Cypress.primaryOriginCommunicator.emit('sync:viewport', viewport)
+      localBus.emit('viewport:changed', viewport, callback)
+    })
+
+    Cypress.primaryOriginCommunicator.on('before:screenshot', (config, originPolicy) => {
+      const callback = () => {
+        Cypress.primaryOriginCommunicator.toSpecBridge(originPolicy, 'before:screenshot:end')
+      }
+
+      handleBeforeScreenshot(config, callback)
+    })
+
+    Cypress.primaryOriginCommunicator.on('url:changed', ({ url }) => {
+      localBus.emit('url:changed', url)
+    })
+
+    Cypress.primaryOriginCommunicator.on('after:screenshot', handleAfterScreenshot)
+
+    const crossOriginLogs = {}
+
+    Cypress.primaryOriginCommunicator.on('log:added', (attrs) => {
+      // Create a new local log representation of the cross origin log.
+      // It will be attached to the current command.
+      // We also keep a reference to it to update it in the future.
+      crossOriginLogs[attrs.id] = Cypress.log(attrs)
+    })
+
+    Cypress.primaryOriginCommunicator.on('log:changed', (attrs) => {
+      // Retrieve the referenced log and update it.
+      const log = crossOriginLogs[attrs.id]
+
+      // this will trigger a log changed event for the log itself.
+      log?.set(attrs)
+    })
```

Comment:
my above suggesting around the logging would/should remove the need to track the origin logs here vs in the primary Cypress instance.

Reply:
I don't think that's true, since we still need to send the logs from the secondary driver to the primary driver and this is what handles that.

Reply:
Were new listeners added to the driver to listen tot he runner's events? If so I missed where that was done; I'm only aways of the driver sending events to the runner which is forwarded to the reporter.